driver/impl/src/main/java/eu/cloudnetservice/driver/impl/network/netty/NettyNetworkHandler.java

```
@@ -76,9 +76,25 @@ public void channelReadComplete(@NonNull ChannelHandlerContext ctx) {
   protected void messageReceived(@NonNull ChannelHandlerContext ctx, @NonNull BasePacket msg) {
     // post directly if the packet has a high priority
     if (msg.prioritized()) {
-      this.doHandlePacket(msg);
+      this.handlePacket(msg);
     } else {
-      this.packetDispatcher().execute(() -> this.doHandlePacket(msg));
+      this.packetDispatcher().execute(() -> this.handlePacket(msg));
+    }
+  }
+
+  /**
+   * Handles the incoming packet and posts it either to the associated waiting query handler or directly into the packet
+   * registry, calling all associated handlers. This method applies exception handling which is not done by
+   * {@link #doHandlePacket(BasePacket)}.
+   *
+   * @param packet the packet hto handle.
+   * @throws NullPointerException if the given packet is null.
+   */
+  private void handlePacket(@NonNull BasePacket packet) {
+    try {
+      this.doHandlePacket(packet);
+    } catch (Exception exception) {
```

I'd go further and catch Throwable instead of silently ignoring OOMEs or similar

---

we should get oracle to have the default UncaughtExceptionHandler call System.exit.
I remember wasting way too many hours because network threads ate my errors and I couldn't for the life of me figure out where.